File: 2211-count-collisions-on-a-road.py
class Solution:
    def countCollisions(self, directions: str) -> int:
        '''
        Those "L" on the left, "R" on the right and "S" in the middle are irrelevant
        '''
        directions = directions.lstrip('L').rstrip('R').replace('S', '')
        return len(directions)


sol = Solution()
print(f'output: {sol.countCollisions("RLRSLL")}, expected: 5')
print(f'output: {sol.countCollisions("LLRR")}, expected: 0')
print(f'output: {sol.countCollisions("SSRSSRLLRSLLRSRSSRLRRRRLLRRLSSRR")}, expected: 20')

File: 2211-count-collisions-on-a-road.java
class CountCollisionsOnARoad {
    public int countCollisions(String directions) {
        int startIndex = 0, endIndex = directions.length() - 1;
        while (startIndex < directions.length() && directions.charAt(startIndex) == 'L') {
            startIndex++;
        }
        while (endIndex >= startIndex && directions.charAt(endIndex) == 'R') {
            endIndex--;
        }
        int res = 0;
        for (int i = startIndex; i <= endIndex; i++) {
            if (directions.charAt(i) == 'S') {
                continue;
            }
            res++;
        }
        return res;
    }
  
    public static void main(String[] args) {
        CountCollisionsOnARoad sol = new CountCollisionsOnARoad();
        System.out.println("Output: " + sol.countCollisions("RLRSLL") + ", expected: 5");
        System.out.println("Output: " + sol.countCollisions("LLRR") + ", expected: 0");
        System.out.println("Output: " + sol.countCollisions("SSRSSRLLRSLLRSRSSRLRRRRLLRRLSSRR") + ", expected: 20");
        System.out.println("Output: " + sol.countCollisions("L") + ", expected: 0");
        System.out.println("Output: " + sol.countCollisions("R") + ", expected: 0");
    }
}
